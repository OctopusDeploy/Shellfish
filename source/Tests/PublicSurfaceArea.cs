using System;
using System.Linq;
using System.Reflection;
using Assent;
using Octopus.Shellfish;
using Xunit;

namespace Tests;

public class PublicSurfaceArea
{
    [Fact]
    public void TheLibraryOnlyExposesWhatWeWantItToExpose()
    {
        var assembly = typeof(ShellExecutor).Assembly;
        var publicMembers =
            from t in assembly.GetExportedTypes()
            from m in t.GetMembers(BindingFlags.Public | BindingFlags.DeclaredOnly | BindingFlags.Static | BindingFlags.Instance)
            where !(m is MethodBase method && method.IsSpecialName)
            select $"{m.DeclaringType?.FullName}.{m.Name}";

        this.Assent(string.Join("\r\n", publicMembers));
    }
}