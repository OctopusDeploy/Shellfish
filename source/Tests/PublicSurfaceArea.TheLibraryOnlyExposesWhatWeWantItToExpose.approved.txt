Octopus.Shellfish.ShellCommandExtensionMethods.WithStdInSource
Octopus.Shellfish.ShellCommandExtensionMethods.WithStdOutTarget
Octopus.Shellfish.ShellCommandExtensionMethods.WithStdErrTarget
Octopus.Shellfish.ShellCommandExtensionMethods.WithStdOutTarget
Octopus.Shellfish.ShellCommandExtensionMethods.WithStdErrTarget
Octopus.Shellfish.IOutputTarget.WriteLine
Octopus.Shellfish.ShellCommand.WithWorkingDirectory
Octopus.Shellfish.ShellCommand.WithArguments
Octopus.Shellfish.ShellCommand.WithArguments
Octopus.Shellfish.ShellCommand.WithEnvironmentVariables
Octopus.Shellfish.ShellCommand.WithCredentials
Octopus.Shellfish.ShellCommand.WithOutputEncoding
Octopus.Shellfish.ShellCommand.WithStdOutTarget
Octopus.Shellfish.ShellCommand.WithStdErrTarget
Octopus.Shellfish.ShellCommand.WithOptions
Octopus.Shellfish.ShellCommand.Execute
Octopus.Shellfish.ShellCommand.ExecuteAsync
Octopus.Shellfish.ShellCommandOptions.value__
Octopus.Shellfish.ShellCommandOptions.None
Octopus.Shellfish.ShellCommandOptions.DoNotThrowOnCancellation
Octopus.Shellfish.ShellCommandResult.ExitCode
Octopus.Shellfish.ShellExecutionException.Errors
Octopus.Shellfish.ShellExecutionException.Message
Octopus.Shellfish.ShellExecutor.ExecuteCommand
Octopus.Shellfish.ShellExecutor.ExecuteCommandWithoutWaiting